gtclang/src/gtclang/Support/ClangCompat/CompilerInvocation.h

```diff
@@ -0,0 +1,53 @@
+//===--------------------------------------------------------------------------------*- C++ -*-===//
+//                         _       _
+//                        | |     | |
+//                    __ _| |_ ___| | __ _ _ __   __ _
+//                   / _` | __/ __| |/ _` | '_ \ / _` |
+//                  | (_| | || (__| | (_| | | | | (_| |
+//                   \__, |\__\___|_|\__,_|_| |_|\__, | - GridTools Clang DSL
+//                    __/ |                       __/ |
+//                   |___/                       |___/
+//
+//
+//  This file is distributed under the MIT License (MIT).
+//  See LICENSE.txt for details.
+//
+//===------------------------------------------------------------------------------------------===//
+
+#ifndef GTCLANG_SUPPORT_CLANGCOMPAT_COMPILER_INVOCATION_H
+#define GTCLANG_SUPPORT_CLANGCOMPAT_COMPILER_INVOCATION_H
+
+#include "clang/Basic/Version.h"
+#include "clang/Frontend/CompilerInvocation.h"
+#include "llvm/Option/Option.h"
+
+namespace gtclang {
+namespace clang_compat {
+namespace CompilerInvocation {
+#if CLANG_VERSION_MAJOR < 9
```

Comment:
Could you add a comment here why this is needed, and why we need it for 8.x and earlier?

Reply:
This was the old implementation. Just the signature changed.

Reply:
I think by fixing the next comment this is obsolete as it is obvious.